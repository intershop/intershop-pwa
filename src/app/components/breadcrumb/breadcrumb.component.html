<div class="breadcrumbs row">
  <ol class="breadcrumbs-list">
    <li><a class="breadcrumbs-list-link breadcrumbs-list-link-home" rel="home" routerLink="/">Home</a>
      <span class="breadcrumb-list-separator">/</span>
    </li>
    <li *ngFor="let url of _urls; let last = last" class="breadcrumbs-list" [ngClass]="{'breadcrumbs-list-active' : last}">
      <a class="breadcrumbs-list-link" *ngIf="!last" (click)="navigateTo(url)" href="{{url}}">{{friendlyName(url)}}</a>
      <span *ngIf="last">{{friendlyName(url)}}</span>
      <span *ngIf="!last" class="breadcrumb-list-separator">/</span>
    </li>
  </ol>
</div>
