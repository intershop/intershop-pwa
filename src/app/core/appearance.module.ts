import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
