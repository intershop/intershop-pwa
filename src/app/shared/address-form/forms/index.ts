import { ADDRESS_FACTORY } from './address-form.factory';

import { AddressDEComponent } from './address-de/address-de.component';
import { AddressDefaultComponent } from './address-default/address-default.component';
import { AddressFormComponent } from './address-form.component';
import { AddressFRComponent } from './address-fr/address-fr.component';
import { AddressGBComponent } from './address-gb/address-gb.component';
import { AddressUSComponent } from './address-us/address-us.component';

import { AddressDEFactory } from './address-de/address-de.factory';
import { AddressDefaultFactory } from './address-default/address-default.factory';
import { AddressFRFactory } from './address-fr/address-fr.factory';
import { AddressGBFactory } from './address-gb/address-gb.factory';
import { AddressUSFactory } from './address-us/address-us.factory';

export const components = [
  AddressFormComponent,
  AddressDefaultComponent,
  AddressDEComponent,
  AddressFRComponent,
  AddressGBComponent,
  AddressUSComponent,
];

const factories = [
  AddressDefaultFactory,
  AddressDEFactory,
  AddressFRFactory,
  AddressGBFactory,
  AddressUSFactory,
];

/**********************************/

export const factoryProviders = factories.map(f => (
  { provide: ADDRESS_FACTORY, useClass: f, multi: true }
));
