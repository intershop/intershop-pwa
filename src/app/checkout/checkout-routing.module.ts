import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { AuthGuard } from '../core/guards/auth.guard';

const routes: Routes = [
  { path: 'basket', loadChildren: './containers/basket-page/basket-page.module#BasketPageModule' },
  {
    path: 'checkout',
    loadChildren: './containers/checkout-page/checkout-page.module#CheckoutPageModule',
    data: { headerType: 'checkout' },
    canActivate: [AuthGuard],
  },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class CheckoutRoutingModule {}
