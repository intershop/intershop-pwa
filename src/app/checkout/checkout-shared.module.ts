import { NgModule } from '@angular/core';
import { SharedProductModule } from '../shared/shared-product.module';
import { SharedModule } from '../shared/shared.module';
import { AddressComponent } from './components/common/address/address.component';
import { BasketAddressSummaryComponent } from './components/common/basket-address-summary/basket-address-summary.component';
import { BasketCostSummaryComponent } from './components/common/basket-cost-summary/basket-cost-summary.component';
import { BasketItemsSummaryComponent } from './components/common/basket-items-summary/basket-items-summary.component';
import { CheckoutProgressBarComponent } from './components/common/checkout-progress-bar/checkout-progress-bar.component';

const sharedComponents = [
  AddressComponent,
  BasketAddressSummaryComponent,
  BasketItemsSummaryComponent,
  BasketCostSummaryComponent,
  CheckoutProgressBarComponent,
];
@NgModule({
  imports: [SharedModule, SharedProductModule],
  declarations: [...sharedComponents],
  exports: [...sharedComponents],
})
export class CheckoutSharedModule {}
