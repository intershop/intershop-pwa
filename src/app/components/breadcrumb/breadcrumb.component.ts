import { Component, OnInit, OnDestroy } from '@angular/core';
import { Router, NavigationEnd } from '@angular/router';
import { BreadcrumbService } from './breadcrumb.service';

/**
 * ng2-breadcrumb reference
 * This component shows a breadcrumb trail for available routes the router can navigate to.
 * It subscribes to the router in order to update the breadcrumb trail as you navigate to a component.
 */
@Component({
  selector: 'is-breadcrumb',
  templateUrl: './breadcrumb.component.html'
})
export class BreadcrumbComponent implements OnInit, OnDestroy {

  public _urls: string[] = [];
  public _routerSubscription: any;

  constructor(
    private router: Router,
    private breadcrumbService: BreadcrumbService
  ) { }

  ngOnInit(): void {
    this._routerSubscription = this.router.events.subscribe((navigationEnd: NavigationEnd) => {
      if (navigationEnd instanceof NavigationEnd) {
        this._urls.length = 0; // Fastest way to clear out array
        this.generateBreadcrumbTrail(navigationEnd.urlAfterRedirects ? navigationEnd.urlAfterRedirects : navigationEnd.url);
      }
    });
  }

  generateBreadcrumbTrail(url: string): void {
    if (!this.breadcrumbService.isRouteHidden(url)) {
      // Add url to beginning of array (since the url is being recursively broken down from full url to its parent)
      this._urls.unshift(url);
    }

    if (url.lastIndexOf('/') > 0) {
      this.generateBreadcrumbTrail(url.substr(0, url.lastIndexOf('/'))); // Find last '/' and add everything before it as a parent route
    }
  }

  navigateTo(url: string): void {
    this.router.navigateByUrl(url);
  }

  friendlyName(url: string): string {
    return !url ? '' : this.breadcrumbService.getFriendlyNameForRoute(url);
  }

  ngOnDestroy(): void {
    if (this._routerSubscription) {
      this._routerSubscription.unsubscribe();
    }
  }
}
