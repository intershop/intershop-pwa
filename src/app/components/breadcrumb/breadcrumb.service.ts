import { Injectable } from '@angular/core';

@Injectable()
export class BreadcrumbService {
  public hideRoutesRegex: any = new Array<string>('/KATEGORIE$', '/category$', '/en_US$', '/USD$', '/de_DE$', '/EUR$'); // To hide these words in breadcrumbs

  /**
   * Returns true if a route should be hidden.
   */
  isRouteHidden(route: string): boolean {
    let hide = false;
    this.hideRoutesRegex.forEach((value: any) => {
      if (new RegExp(value).exec(route)) {
        hide = true;
      }
    });
    return hide;
  }
}
