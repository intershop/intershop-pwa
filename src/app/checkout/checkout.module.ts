import { NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { CheckoutRoutingModule } from './checkout-routing.module';
import { BasketService } from './services/basket/basket.service';
import { checkoutEffects, checkoutReducers } from './store/checkout.system';

@NgModule({
  imports: [
    CheckoutRoutingModule,
    StoreModule.forFeature('checkout', checkoutReducers),
    EffectsModule.forFeature(checkoutEffects),
  ],
  providers: [BasketService],
})
export class CheckoutModule {}
