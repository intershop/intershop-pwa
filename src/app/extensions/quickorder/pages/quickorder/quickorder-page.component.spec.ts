import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';
import { MockComponent } from 'ng-mocks';
import { instance, mock } from 'ts-mockito';

import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
import { BreadcrumbComponent } from 'ish-shared/components/common/breadcrumb/breadcrumb.component';
import { LoadingComponent } from 'ish-shared/components/common/loading/loading.component';
import { InputComponent } from 'ish-shared/forms/components/input/input.component';

import { QuickorderPageComponent } from './quickorder-page.component';

describe('Quickorder Page Component', () => {
  let component: QuickorderPageComponent;
  let fixture: ComponentFixture<QuickorderPageComponent>;
  let element: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, TranslateModule.forRoot()],
      declarations: [
        MockComponent(BreadcrumbComponent),
        MockComponent(InputComponent),
        MockComponent(LoadingComponent),
        QuickorderPageComponent,
      ],
      providers: [
        { provide: ShoppingFacade, useFactory: () => instance(mock(ShoppingFacade)) },
        { provide: CheckoutFacade, useFactory: () => instance(mock(CheckoutFacade)) },
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(QuickorderPageComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
  });

  it('should be always initialized with numberOfRows quick order lines', () => {
    fixture.detectChanges();

    expect(component.quickOrderlines).toHaveLength(component.numberOfRows);
  });

  it('should always delete one line', () => {
    fixture.detectChanges();
    component.deleteItem(0);
    expect(component.quickOrderlines).toHaveLength(component.numberOfRows - 1);
  });

  it('should always add one line', () => {
    fixture.detectChanges();
    component.addRows(1);
    expect(component.quickOrderlines).toHaveLength(component.numberOfRows + 1);
  });
});
