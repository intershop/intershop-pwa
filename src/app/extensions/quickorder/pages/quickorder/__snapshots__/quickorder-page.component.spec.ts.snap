// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quickorder Page Component should be created 1`] = `
<form novalidate="" class="ng-untouched ng-pristine ng-invalid">
  <ish-breadcrumb></ish-breadcrumb>
  <div class="row">
    <div class="col-md-12"><h1>quickorder.page.title</h1></div>
  </div>
  <div class="row">
    <div class="col-md-8">
      <h3>quickorder.page.subtitle</h3>
      <p>quickorder.page.subtitle.description</p>
      <div class="list-header d-md-flex">
        <div class="col-md-6 list-header-item no-seperator">quickorder.page.product.id</div>
        <div class="col-md-2 list-header-item no-seperator">quickorder.page.product.quantity</div>
      </div>
      <div class="pt-1">
        <div
          formarrayname="quickOrderlines"
          ng-reflect-name="quickOrderlines"
          class="ng-untouched ng-pristine ng-invalid"
        >
          <div
            class="row list-item-row py-2 ng-untouched ng-pristine ng-invalid"
            ng-reflect-name="0"
          >
            <div class="col-sm-6 list-item search-container">
              <div class="d-sm-none">quickorder.page.product.id</div>
              <div class="quickorder-search-container position-relative">
                <input
                  class="form-control w-100 col-12 ng-untouched ng-pristine ng-valid"
                  formcontrolname="sku"
                  type="text"
                  ng-reflect-name="sku"
                />
              </div>
            </div>
            <div class="col-sm-2 list-item">
              <div class="d-sm-none">quickorder.page.product.quantity</div>
              <ish-input
                controlname="quantity"
                inputclass="w-100"
                min="1"
                type="number"
                ng-reflect-control-name="quantity"
                ng-reflect-input-class="w-100"
                ng-reflect-type="number"
                ng-reflect-min="1"
              ></ish-input>
            </div>
            <div class="col-sm-2 list-item">
              <a class="btn btn-link remove-row" title="remove-current-row"
                >quickorder.page.remove.row</a
              >
            </div>
          </div>
          <div
            class="row list-item-row py-2 ng-untouched ng-pristine ng-invalid"
            ng-reflect-name="1"
          >
            <div class="col-sm-6 list-item search-container">
              <div class="d-sm-none">quickorder.page.product.id</div>
              <div class="quickorder-search-container position-relative">
                <input
                  class="form-control w-100 col-12 ng-untouched ng-pristine ng-valid"
                  formcontrolname="sku"
                  type="text"
                  ng-reflect-name="sku"
                />
              </div>
            </div>
            <div class="col-sm-2 list-item">
              <div class="d-sm-none">quickorder.page.product.quantity</div>
              <ish-input
                controlname="quantity"
                inputclass="w-100"
                min="1"
                type="number"
                ng-reflect-control-name="quantity"
                ng-reflect-input-class="w-100"
                ng-reflect-type="number"
                ng-reflect-min="1"
              ></ish-input>
            </div>
            <div class="col-sm-2 list-item">
              <a class="btn btn-link remove-row" title="remove-current-row"
                >quickorder.page.remove.row</a
              >
            </div>
          </div>
          <div
            class="row list-item-row py-2 ng-untouched ng-pristine ng-invalid"
            ng-reflect-name="2"
          >
            <div class="col-sm-6 list-item search-container">
              <div class="d-sm-none">quickorder.page.product.id</div>
              <div class="quickorder-search-container position-relative">
                <input
                  class="form-control w-100 col-12 ng-untouched ng-pristine ng-valid"
                  formcontrolname="sku"
                  type="text"
                  ng-reflect-name="sku"
                />
              </div>
            </div>
            <div class="col-sm-2 list-item">
              <div class="d-sm-none">quickorder.page.product.quantity</div>
              <ish-input
                controlname="quantity"
                inputclass="w-100"
                min="1"
                type="number"
                ng-reflect-control-name="quantity"
                ng-reflect-input-class="w-100"
                ng-reflect-type="number"
                ng-reflect-min="1"
              ></ish-input>
            </div>
            <div class="col-sm-2 list-item">
              <a class="btn btn-link remove-row" title="remove-current-row"
                >quickorder.page.remove.row</a
              >
            </div>
          </div>
          <div
            class="row list-item-row py-2 ng-untouched ng-pristine ng-invalid"
            ng-reflect-name="3"
          >
            <div class="col-sm-6 list-item search-container">
              <div class="d-sm-none">quickorder.page.product.id</div>
              <div class="quickorder-search-container position-relative">
                <input
                  class="form-control w-100 col-12 ng-untouched ng-pristine ng-valid"
                  formcontrolname="sku"
                  type="text"
                  ng-reflect-name="sku"
                />
              </div>
            </div>
            <div class="col-sm-2 list-item">
              <div class="d-sm-none">quickorder.page.product.quantity</div>
              <ish-input
                controlname="quantity"
                inputclass="w-100"
                min="1"
                type="number"
                ng-reflect-control-name="quantity"
                ng-reflect-input-class="w-100"
                ng-reflect-type="number"
                ng-reflect-min="1"
              ></ish-input>
            </div>
            <div class="col-sm-2 list-item">
              <a class="btn btn-link remove-row" title="remove-current-row"
                >quickorder.page.remove.row</a
              >
            </div>
          </div>
          <div
            class="row list-item-row py-2 ng-untouched ng-pristine ng-invalid"
            ng-reflect-name="4"
          >
            <div class="col-sm-6 list-item search-container">
              <div class="d-sm-none">quickorder.page.product.id</div>
              <div class="quickorder-search-container position-relative">
                <input
                  class="form-control w-100 col-12 ng-untouched ng-pristine ng-valid"
                  formcontrolname="sku"
                  type="text"
                  ng-reflect-name="sku"
                />
              </div>
            </div>
            <div class="col-sm-2 list-item">
              <div class="d-sm-none">quickorder.page.product.quantity</div>
              <ish-input
                controlname="quantity"
                inputclass="w-100"
                min="1"
                type="number"
                ng-reflect-control-name="quantity"
                ng-reflect-input-class="w-100"
                ng-reflect-type="number"
                ng-reflect-min="1"
              ></ish-input>
            </div>
            <div class="col-sm-2 list-item">
              <a class="btn btn-link remove-row" title="remove-current-row"
                >quickorder.page.remove.row</a
              >
            </div>
          </div>
          <a class="d-block">quickorder.page.add.row</a>
          <hr class="mt-2 mb-4" />
        </div>
      </div>
    </div>
  </div>
  <div class="btn-group">
    <button class="btn btn-primary" type="submit">quickorder.page.add.cart</button
    ><button class="btn btn-link" type="reset">quickorder.page.reset.form</button>
  </div>
  <div class="row mt-4">
    <div class="col-md-4">
      <h3>quickorder.page.csv.title</h3>
      <p>quickorder.page.csv.subtitle</p>
      <form novalidate="" class="ng-untouched ng-pristine ng-invalid">
        <div class="form-group section section-seperator">
          <input
            accept=".csv"
            class="form-control"
            id="CSVFile"
            name="csvFile"
            size="35"
            type="file"
          />
          <div class="position-absolute" ng-reflect-ng-switch="0"><small></small></div>
        </div>
        <div>
          <button class="btn btn-primary" name="addToCartCSV" type="submit">
            quickorder.page.add.cart</button
          ><button class="btn btn-link" name="reset" type="reset">
            quickorder.page.csv.reset.form
          </button>
        </div>
      </form>
    </div>
  </div>
</form>
`;
